### PR TITLE
Reorganize source files

### LINKED ISSUE #1068: Reorganize file structure in ArborX
Current:
```
src
├── ArborX.hpp
├── ArborX_BruteForce.hpp
├── ArborX_Config.hpp.in
├── ArborX_CrsGraphWrapper.hpp
├── ArborX_DBSCAN.hpp
├── ArborX_DistributedTree.hpp
├── ArborX_HDBSCAN.hpp
├── ArborX_LinearBVH.hpp
├── ArborX_Version.hpp.in
├── details
│   ├── ArborX_AccessTraits.hpp
│   ├── ArborX_AttachIndices.hpp
│   ├── ArborX_Callbacks.hpp
│   ├── ArborX_Dendrogram.hpp
│   ├── ArborX_DetailsBatchedQueries.hpp
│   ├── ArborX_DetailsBruteForceImpl.hpp
│   ├── ArborX_DetailsCartesianGrid.hpp
│   ├── ArborX_DetailsContainers.hpp
│   ├── ArborX_DetailsCrsGraphWrapperImpl.hpp
│   ├── ArborX_DetailsDendrogram.hpp
│   ├── ArborX_DetailsDistributedTreeImpl.hpp
│   ├── ArborX_DetailsDistributedTreeNearest.hpp
│   ├── ArborX_DetailsDistributedTreeSpatial.hpp
│   ├── ArborX_DetailsDistributedTreeUtils.hpp
│   ├── ArborX_DetailsDistributor.hpp
│   ├── ArborX_DetailsExpandHalfToFull.hpp
│   ├── ArborX_DetailsFDBSCAN.hpp
│   ├── ArborX_DetailsFDBSCANDenseBox.hpp
│   ├── ArborX_DetailsHalfTraversal.hpp
│   ├── ArborX_DetailsHappyTreeFriends.hpp
│   ├── ArborX_DetailsHeap.hpp
│   ├── ArborX_DetailsLegacy.hpp
│   ├── ArborX_DetailsMinimumSpanningTree.hpp
│   ├── ArborX_DetailsMortonCode.hpp
│   ├── ArborX_DetailsMutualReachabilityDistance.hpp
│   ├── ArborX_DetailsNearestBufferProvider.hpp
│   ├── ArborX_DetailsNode.hpp
│   ├── ArborX_DetailsOperatorFunctionObjects.hpp
│   ├── ArborX_DetailsPermutedData.hpp
│   ├── ArborX_DetailsPriorityQueue.hpp
│   ├── ArborX_DetailsSortUtils.hpp
│   ├── ArborX_DetailsStack.hpp
│   ├── ArborX_DetailsTreeConstruction.hpp
│   ├── ArborX_DetailsTreeNodeLabeling.hpp
│   ├── ArborX_DetailsTreeTraversal.hpp
│   ├── ArborX_DetailsTreeVisualization.hpp
│   ├── ArborX_DetailsUnionFind.hpp
│   ├── ArborX_DetailsUtils.hpp
│   ├── ArborX_DetailsWeightedEdge.hpp
│   ├── ArborX_Exception.hpp
│   ├── ArborX_IndexableGetter.hpp
│   ├── ArborX_MinimumSpanningTree.hpp
│   ├── ArborX_NeighborList.hpp
│   ├── ArborX_PairIndexRank.hpp
│   ├── ArborX_PairValueIndex.hpp
│   ├── ArborX_PredicateHelpers.hpp
│   ├── ArborX_Predicates.hpp
│   ├── ArborX_SpaceFillingCurves.hpp
│   └── ArborX_TraversalPolicy.hpp
├── geometry
│   ├── ArborX_Box.hpp
│   ├── ArborX_DetailsAlgorithms.hpp
│   ├── ArborX_GeometryTraits.hpp
│   ├── ArborX_HyperBox.hpp
│   ├── ArborX_HyperPoint.hpp
│   ├── ArborX_HyperSphere.hpp
│   ├── ArborX_HyperTriangle.hpp
│   ├── ArborX_KDOP.hpp
│   ├── ArborX_Point.hpp
│   ├── ArborX_Ray.hpp
│   └── ArborX_Sphere.hpp
├── interpolation
│   ├── ArborX_InterpMovingLeastSquares.hpp
│   └── details
│       ├── ArborX_InterpDetailsCompactRadialBasisFunction.hpp
│       ├── ArborX_InterpDetailsMovingLeastSquaresCoefficients.hpp
│       ├── ArborX_InterpDetailsPolynomialBasis.hpp
│       └── ArborX_InterpDetailsSymmetricPseudoInverseSVD.hpp
├── kokkos_ext
│   ├── ArborX_DetailsKokkosExtAccessibilityTraits.hpp
│   ├── ArborX_DetailsKokkosExtArithmeticTraits.hpp
│   ├── ArborX_DetailsKokkosExtMinMaxOperations.hpp
│   ├── ArborX_DetailsKokkosExtMinMaxReduce.hpp
│   ├── ArborX_DetailsKokkosExtSort.hpp
│   ├── ArborX_DetailsKokkosExtStdAlgorithms.hpp
│   ├── ArborX_DetailsKokkosExtSwap.hpp
│   ├── ArborX_DetailsKokkosExtVersion.hpp
│   └── ArborX_DetailsKokkosExtViewHelpers.hpp
└── original
```

Proposed:
```
src
├── ArborX.hpp
├── ArborX_Config.hpp.in
├── ArborX_Version.hpp.in
├── cluster
│   ├── ArborX_DBSCAN.hpp
│   ├── ArborX_Dendrogram.hpp
│   ├── ArborX_HDBSCAN.hpp
│   ├── ArborX_MinimumSpanningTree.hpp
│   └── details
│       ├── ArborX_DetailsDendrogram.hpp
│       ├── ArborX_DetailsFDBSCAN.hpp
│       ├── ArborX_DetailsFDBSCANDenseBox.hpp
│       ├── ArborX_DetailsMinimumSpanningTree.hpp
│       ├── ArborX_DetailsMutualReachabilityDistance.hpp
│       ├── ArborX_DetailsUnionFind.hpp
│       └── ArborX_DetailsWeightedEdge.hpp
├── geometry
│   ├── ArborX_Box.hpp
│   ├── ArborX_DetailsAlgorithms.hpp
│   ├── ArborX_GeometryTraits.hpp
│   ├── ArborX_HyperBox.hpp
│   ├── ArborX_HyperPoint.hpp
│   ├── ArborX_HyperSphere.hpp
│   ├── ArborX_HyperTriangle.hpp
│   ├── ArborX_KDOP.hpp
│   ├── ArborX_Point.hpp
│   ├── ArborX_Ray.hpp
│   └── ArborX_Sphere.hpp
├── interpolation
│   ├── ArborX_InterpMovingLeastSquares.hpp
│   └── details
│       ├── ArborX_InterpDetailsCompactRadialBasisFunction.hpp
│       ├── ArborX_InterpDetailsMovingLeastSquaresCoefficients.hpp
│       ├── ArborX_InterpDetailsPolynomialBasis.hpp
│       └── ArborX_InterpDetailsSymmetricPseudoInverseSVD.hpp
├── kokkos_ext
│   ├── ArborX_DetailsKokkosExtAccessibilityTraits.hpp
│   ├── ArborX_DetailsKokkosExtArithmeticTraits.hpp
│   ├── ArborX_DetailsKokkosExtMinMaxOperations.hpp
│   ├── ArborX_DetailsKokkosExtMinMaxReduce.hpp
│   ├── ArborX_DetailsKokkosExtSort.hpp
│   ├── ArborX_DetailsKokkosExtStdAlgorithms.hpp
│   ├── ArborX_DetailsKokkosExtSwap.hpp
│   ├── ArborX_DetailsKokkosExtVersion.hpp
│   └── ArborX_DetailsKokkosExtViewHelpers.hpp
└── spatial
    ├── ArborX_BruteForce.hpp
    ├── ArborX_CrsGraphWrapper.hpp
    ├── ArborX_DistributedTree.hpp
    ├── ArborX_LinearBVH.hpp
    └── details
        ├── ArborX_AccessTraits.hpp
        ├── ArborX_AttachIndices.hpp
        ├── ArborX_Callbacks.hpp
        ├── ArborX_DetailsBatchedQueries.hpp
        ├── ArborX_DetailsBruteForceImpl.hpp
        ├── ArborX_DetailsCartesianGrid.hpp
        ├── ArborX_DetailsContainers.hpp
        ├── ArborX_DetailsCrsGraphWrapperImpl.hpp
        ├── ArborX_DetailsDistributedTreeImpl.hpp
        ├── ArborX_DetailsDistributedTreeNearest.hpp
        ├── ArborX_DetailsDistributedTreeSpatial.hpp
        ├── ArborX_DetailsDistributedTreeUtils.hpp
        ├── ArborX_DetailsDistributor.hpp
        ├── ArborX_DetailsExpandHalfToFull.hpp
        ├── ArborX_DetailsHalfTraversal.hpp
        ├── ArborX_DetailsHappyTreeFriends.hpp
        ├── ArborX_DetailsHeap.hpp
        ├── ArborX_DetailsLegacy.hpp
        ├── ArborX_DetailsMortonCode.hpp
        ├── ArborX_DetailsNearestBufferProvider.hpp
        ├── ArborX_DetailsNode.hpp
        ├── ArborX_DetailsOperatorFunctionObjects.hpp
        ├── ArborX_DetailsPermutedData.hpp
        ├── ArborX_DetailsPriorityQueue.hpp
        ├── ArborX_DetailsSortUtils.hpp
        ├── ArborX_DetailsStack.hpp
        ├── ArborX_DetailsTreeConstruction.hpp
        ├── ArborX_DetailsTreeNodeLabeling.hpp
        ├── ArborX_DetailsTreeTraversal.hpp
        ├── ArborX_DetailsTreeVisualization.hpp
        ├── ArborX_DetailsUtils.hpp
        ├── ArborX_Exception.hpp
        ├── ArborX_IndexableGetter.hpp
        ├── ArborX_NeighborList.hpp
        ├── ArborX_PairIndexRank.hpp
        ├── ArborX_PairValueIndex.hpp
        ├── ArborX_PredicateHelpers.hpp
        ├── ArborX_Predicates.hpp
        ├── ArborX_SpaceFillingCurves.hpp
        └── ArborX_TraversalPolicy.hpp
```

Not 100% sure what to do with tests. I would not want to duplicate the infrastructure.

### PR DESCRIPTION
**Pull request by aprokop**
Fix #1068.

### CODE REVIEW

**Comment by aprokop** [comment]
```
src
├── ArborX.hpp
├── ArborX_Config.hpp.in
├── ArborX_Version.hpp.in
├── cluster
│   ├── ArborX_DBSCAN.hpp
│   ├── ArborX_Dendrogram.hpp
│   ├── ArborX_HDBSCAN.hpp
│   ├── ArborX_MinimumSpanningTree.hpp
│   └── details
│       ├── ArborX_DetailsCartesianGrid.hpp
│       ├── ArborX_DetailsDendrogram.hpp
│       ├── ArborX_DetailsFDBSCAN.hpp
│       ├── ArborX_DetailsFDBSCANDenseBox.hpp
│       ├── ArborX_DetailsMinimumSpanningTree.hpp
│       ├── ArborX_DetailsMutualReachabilityDistance.hpp
│       ├── ArborX_DetailsUnionFind.hpp
│       └── ArborX_DetailsWeightedEdge.hpp
├── distributed
│   ├── ArborX_DistributedTree.hpp
│   └── details
│       ├── ArborX_DetailsDistributedTreeImpl.hpp
│       ├── ArborX_DetailsDistributedTreeNearest.hpp
│       ├── ArborX_DetailsDistributedTreeNearestHelpers.hpp
│       ├── ArborX_DetailsDistributedTreeSpatial.hpp
│       ├── ArborX_DetailsDistributedTreeUtils.hpp
│       └── ArborX_DetailsDistributor.hpp
├── geometry
│   ├── ArborX_Box.hpp
│   ├── ArborX_DetailsAlgorithms.hpp
│   ├── ArborX_GeometryTraits.hpp
│   ├── ArborX_KDOP.hpp
│   ├── ArborX_Point.hpp
│   ├── ArborX_Ray.hpp
│   ├── ArborX_Segment.hpp
│   ├── ArborX_Sphere.hpp
│   ├── ArborX_Tetrahedron.hpp
│   └── ArborX_Triangle.hpp
├── interpolation
│   ├── ArborX_InterpMovingLeastSquares.hpp
│   └── details
│       ├── ArborX_InterpDetailsCompactRadialBasisFunction.hpp
│       ├── ArborX_InterpDetailsMovingLeastSquaresCoefficients.hpp
│       ├── ArborX_InterpDetailsPolynomialBasis.hpp
│       └── ArborX_InterpDetailsSymmetricPseudoInverseSVD.hpp
├── kokkos_ext
│   ├── ArborX_DetailsKokkosExtAccessibilityTraits.hpp
│   ├── ArborX_DetailsKokkosExtArithmeticTraits.hpp
│   ├── ArborX_DetailsKokkosExtKernelStdAlgorithms.hpp
│   ├── ArborX_DetailsKokkosExtMinMaxReduce.hpp
│   ├── ArborX_DetailsKokkosExtSort.hpp
│   ├── ArborX_DetailsKokkosExtStdAlgorithms.hpp
│   ├── ArborX_DetailsKokkosExtUninitializedMemoryAlgorithms.hpp
│   ├── ArborX_DetailsKokkosExtVersion.hpp
│   └── ArborX_DetailsKokkosExtViewHelpers.hpp
├── misc
│   ├── ArborX_DetailsContainers.hpp
│   ├── ArborX_DetailsHeap.hpp
│   ├── ArborX_DetailsMortonCode.hpp
│   ├── ArborX_DetailsOperatorFunctionObjects.hpp
│   ├── ArborX_DetailsPriorityQueue.hpp
│   ├── ArborX_DetailsSortUtils.hpp
│   ├── ArborX_DetailsStack.hpp
│   ├── ArborX_DetailsUtils.hpp
│   ├── ArborX_DetailsVector.hpp
│   ├── ArborX_Exception.hpp
│   └── ArborX_SpaceFillingCurves.hpp
└── spatial
    ├── ArborX_BruteForce.hpp
    ├── ArborX_CrsGraphWrapper.hpp
    ├── ArborX_LinearBVH.hpp
    └── details
        ├── ArborX_AccessTraits.hpp
        ├── ArborX_AttachIndices.hpp
        ├── ArborX_Callbacks.hpp
        ├── ArborX_DetailsBatchedQueries.hpp
        ├── ArborX_DetailsBruteForceImpl.hpp
        ├── ArborX_DetailsCrsGraphWrapperImpl.hpp
        ├── ArborX_DetailsExpandHalfToFull.hpp
        ├── ArborX_DetailsHalfTraversal.hpp
        ├── ArborX_DetailsHappyTreeFriends.hpp
        ├── ArborX_DetailsNearestBufferProvider.hpp
        ├── ArborX_DetailsNode.hpp
        ├── ArborX_DetailsPermutedData.hpp
        ├── ArborX_DetailsTreeConstruction.hpp
        ├── ArborX_DetailsTreeNodeLabeling.hpp
        ├── ArborX_DetailsTreeTraversal.hpp
        ├── ArborX_DetailsTreeVisualization.hpp
        ├── ArborX_IndexableGetter.hpp
        ├── ArborX_NeighborList.hpp
        ├── ArborX_PairValueIndex.hpp
        ├── ArborX_PredicateHelpers.hpp
        ├── ArborX_Predicates.hpp
        └── ArborX_TraversalPolicy.hpp
```

**Comment by dalg24** [comment]
I would put `ArborX_SpaceFillingCurves.hpp` and `ArborX_MortonCode.hpp` in `spatial/`

File: CMakeLists.txt [hunk 1]
```diff
@@ -73,18 +73,30 @@ cmake_dependent_option(ARBORX_ENABLE_GPU_AWARE_MPI
 
 target_include_directories(ArborX INTERFACE
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/details>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/geometry>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/kokkos_ext>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/cluster>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/cluster/details>
```
Comment (dalg24) [inline CMakeLists.txt:79]: I consider this a mistake when I introduced a details/ subdirectory and enabled including w/o naming the path.
I wonder if that would not be a good occasion to drop the `Details` in the file names.
That is `#include <ArborX_DetailsFoo.hpp>` renamed to `<details/ArborX_Foo.hpp>`
Thoughts?

Comment (aprokop) [reply CMakeLists.txt:79]: This is fine with me. Should we rename `details/` to `impl/`?

Comment (dalg24) [reply CMakeLists.txt:79]: I am tempted to say detail[s] singular.  I don't particularly like "impl".
Guarding like Kokkos did is a pain.  I would rather make sure we clearly document what users should include and consider whether to provide an IWYU mapping file.

Comment (aprokop) [reply CMakeLists.txt:79]: The same should probably be done with `kokkos_ext` and `misc`, so as we see ```
#include <kokkos_ext/<blah>`
#include <misc/blah>
```

**Comment by aprokop** [comment]
Tree as of d84af63a0a5f14fe6c14d50153fabfd47b019a30:
```
src/
├── ArborX.hpp
├── ArborX_Config.hpp.in
├── ArborX_Version.hpp.in
├── cluster
│   ├── ArborX_DBSCAN.hpp
│   ├── ArborX_Dendrogram.hpp
│   ├── ArborX_HDBSCAN.hpp
│   ├── ArborX_MinimumSpanningTree.hpp
│   └── details
│       ├── ArborX_CartesianGrid.hpp
│       ├── ArborX_Dendrogram.hpp
│       ├── ArborX_FDBSCAN.hpp
│       ├── ArborX_FDBSCANDenseBox.hpp
│       ├── ArborX_MinimumSpanningTree.hpp
│       ├── ArborX_MutualReachabilityDistance.hpp
│       ├── ArborX_UnionFind.hpp
│       └── ArborX_WeightedEdge.hpp
├── distributed
│   ├── ArborX_DistributedTree.hpp
│   └── details
│       ├── ArborX_DistributedTreeImpl.hpp
│       ├── ArborX_DistributedTreeNearest.hpp
│       ├── ArborX_DistributedTreeNearestHelpers.hpp
│       ├── ArborX_DistributedTreeSpatial.hpp
│       ├── ArborX_DistributedTreeUtils.hpp
│       └── ArborX_Distributor.hpp
├── geometry
│   ├── ArborX_Box.hpp
│   ├── ArborX_GeometryTraits.hpp
│   ├── ArborX_KDOP.hpp
│   ├── ArborX_Point.hpp
│   ├── ArborX_Ray.hpp
│   ├── ArborX_Segment.hpp
│   ├── ArborX_Sphere.hpp
│   ├── ArborX_Tetrahedron.hpp
│   ├── ArborX_Triangle.hpp
│   └── details
│       └── ArborX_Algorithms.hpp
├── interpolation
│   ├── ArborX_InterpMovingLeastSquares.hpp
│   └── details
│       ├── ArborX_InterpDetailsCompactRadialBasisFunction.hpp
│       ├── ArborX_InterpDetailsMovingLeastSquaresCoefficients.hpp
│       ├── ArborX_InterpDetailsPolynomialBasis.hpp
│       └── ArborX_InterpDetailsSymmetricPseudoInverseSVD.hpp
├── kokkos_ext
│   ├── ArborX_KokkosExtAccessibilityTraits.hpp
│   ├── ArborX_KokkosExtArithmeticTraits.hpp
│   ├── ArborX_KokkosExtKernelStdAlgorithms.hpp
│   ├── ArborX_KokkosExtMinMaxReduce.hpp
│   ├── ArborX_KokkosExtSort.hpp
│   ├── ArborX_KokkosExtStdAlgorithms.hpp
│   ├── ArborX_KokkosExtUninitializedMemoryAlgorithms.hpp
│   ├── ArborX_KokkosExtVersion.hpp
│   └── ArborX_KokkosExtViewHelpers.hpp
├── misc
│   ├── ArborX_Containers.hpp
│   ├── ArborX_Exception.hpp
│   ├── ArborX_Heap.hpp
│   ├── ArborX_OperatorFunctionObjects.hpp
│   ├── ArborX_PriorityQueue.hpp
│   ├── ArborX_SortUtils.hpp
│   ├── ArborX_Stack.hpp
│   ├── ArborX_Utils.hpp
│   └── ArborX_Vector.hpp
└── spatial
    ├── ArborX_BruteForce.hpp
    ├── ArborX_CrsGraphWrapper.hpp
    ├── ArborX_LinearBVH.hpp
    └── details
        ├── ArborX_AccessTraits.hpp
        ├── ArborX_AttachIndices.hpp
        ├── ArborX_BatchedQueries.hpp
        ├── ArborX_BruteForceImpl.hpp
        ├── ArborX_Callbacks.hpp
        ├── ArborX_CrsGraphWrapperImpl.hpp
        ├── ArborX_ExpandHalfToFull.hpp
        ├── ArborX_HalfTraversal.hpp
        ├── ArborX_HappyTreeFriends.hpp
        ├── ArborX_IndexableGetter.hpp
        ├── ArborX_MortonCode.hpp
        ├── ArborX_NearestBufferProvider.hpp
        ├── ArborX_NeighborList.hpp
        ├── ArborX_Node.hpp
        ├── ArborX_PairValueIndex.hpp
        ├── ArborX_PermutedData.hpp
        ├── ArborX_PredicateHelpers.hpp
        ├── ArborX_Predicates.hpp
        ├── ArborX_SpaceFillingCurves.hpp
        ├── ArborX_TraversalPolicy.hpp
        ├── ArborX_TreeConstruction.hpp
        ├── ArborX_TreeNodeLabeling.hpp
        ├── ArborX_TreeTraversal.hpp
        └── ArborX_TreeVisualization.hpp

13 directories, 82 files
```

**Comment by aprokop** [comment]
Irrelevant build failure (no CUDA-capable device is detected) for CUDA-11.7.1 build.

**Comment by aprokop** [comment]
Status as of 900702282eec19d7dca07abb7220b07bbaf40027:
```
src/
├── ArborX.hpp
├── ArborX_Config.hpp.in
├── ArborX_Version.hpp.in
├── cluster
│   ├── ArborX_DBSCAN.hpp
│   ├── ArborX_Dendrogram.hpp
│   ├── ArborX_HDBSCAN.hpp
│   ├── ArborX_MinimumSpanningTree.hpp
│   └── detail
│       ├── ArborX_CartesianGrid.hpp
│       ├── ArborX_Dendrogram.hpp
│       ├── ArborX_FDBSCAN.hpp
│       ├── ArborX_FDBSCANDenseBox.hpp
│       ├── ArborX_MinimumSpanningTree.hpp
│       ├── ArborX_MutualReachabilityDistance.hpp
│       ├── ArborX_UnionFind.hpp
│       └── ArborX_WeightedEdge.hpp
├── distributed
│   ├── ArborX_DistributedTree.hpp
│   └── detail
│       ├── ArborX_DistributedTreeImpl.hpp
│       ├── ArborX_DistributedTreeNearest.hpp
│       ├── ArborX_DistributedTreeNearestHelpers.hpp
│       ├── ArborX_DistributedTreeSpatial.hpp
│       ├── ArborX_DistributedTreeUtils.hpp
│       └── ArborX_Distributor.hpp
├── geometry
│   ├── ArborX_Box.hpp
│   ├── ArborX_GeometryTraits.hpp
│   ├── ArborX_KDOP.hpp
│   ├── ArborX_Point.hpp
│   ├── ArborX_Ray.hpp
│   ├── ArborX_Segment.hpp
│   ├── ArborX_Sphere.hpp
│   ├── ArborX_Tetrahedron.hpp
│   ├── ArborX_Triangle.hpp
│   └── detail
│       └── ArborX_Algorithms.hpp
├── interpolation
│   ├── ArborX_InterpMovingLeastSquares.hpp
│   └── detail
│       ├── ArborX_InterpDetailsCompactRadialBasisFunction.hpp
│       ├── ArborX_InterpDetailsMovingLeastSquaresCoefficients.hpp
│       ├── ArborX_InterpDetailsPolynomialBasis.hpp
│       └── ArborX_InterpDetailsSymmetricPseudoInverseSVD.hpp
├── kokkos_ext
│   ├── ArborX_KokkosExtAccessibilityTraits.hpp
│   ├── ArborX_KokkosExtArithmeticTraits.hpp
│   ├── ArborX_KokkosExtKernelStdAlgorithms.hpp
│   ├── ArborX_KokkosExtMinMaxReduce.hpp
│   ├── ArborX_KokkosExtSort.hpp
│   ├── ArborX_KokkosExtStdAlgorithms.hpp
│   ├── ArborX_KokkosExtUninitializedMemoryAlgorithms.hpp
│   ├── ArborX_KokkosExtVersion.hpp
│   └── ArborX_KokkosExtViewHelpers.hpp
├── misc
│   ├── ArborX_Containers.hpp
│   ├── ArborX_Exception.hpp
│   ├── ArborX_Heap.hpp
│   ├── ArborX_OperatorFunctionObjects.hpp
│   ├── ArborX_PriorityQueue.hpp
│   ├── ArborX_SortUtils.hpp
│   ├── ArborX_Stack.hpp
│   ├── ArborX_Utils.hpp
│   └── ArborX_Vector.hpp
└── spatial
    ├── ArborX_BruteForce.hpp
    ├── ArborX_CrsGraphWrapper.hpp
    ├── ArborX_LinearBVH.hpp
    └── detail
        ├── ArborX_AccessTraits.hpp
        ├── ArborX_AttachIndices.hpp
        ├── ArborX_BatchedQueries.hpp
        ├── ArborX_BruteForceImpl.hpp
        ├── ArborX_Callbacks.hpp
        ├── ArborX_CrsGraphWrapperImpl.hpp
        ├── ArborX_ExpandHalfToFull.hpp
        ├── ArborX_HalfTraversal.hpp
        ├── ArborX_HappyTreeFriends.hpp
        ├── ArborX_IndexableGetter.hpp
        ├── ArborX_MortonCode.hpp
        ├── ArborX_NearestBufferProvider.hpp
        ├── ArborX_NeighborList.hpp
        ├── ArborX_Node.hpp
        ├── ArborX_PairValueIndex.hpp
        ├── ArborX_PermutedData.hpp
        ├── ArborX_PredicateHelpers.hpp
        ├── ArborX_Predicates.hpp
        ├── ArborX_SpaceFillingCurves.hpp
        ├── ArborX_TraversalPolicy.hpp
        ├── ArborX_TreeConstruction.hpp
        ├── ArborX_TreeNodeLabeling.hpp
        ├── ArborX_TreeTraversal.hpp
        └── ArborX_TreeVisualization.hpp
```

**Comment by aprokop** [comment]
Status as of 9cdee606969945a3544b9b3a39cf6cf29b7ce6a0:
```
src/
├── ArborX.hpp
├── ArborX_Config.hpp.in
├── ArborX_Version.hpp.in
├── cluster
│   ├── ArborX_DBSCAN.hpp
│   ├── ArborX_Dendrogram.hpp
│   ├── ArborX_HDBSCAN.hpp
│   ├── ArborX_MinimumSpanningTree.hpp
│   └── detail
│       ├── ArborX_CartesianGrid.hpp
│       ├── ArborX_Dendrogram.hpp
│       ├── ArborX_FDBSCAN.hpp
│       ├── ArborX_FDBSCANDenseBox.hpp
│       ├── ArborX_MinimumSpanningTree.hpp
│       ├── ArborX_MutualReachabilityDistance.hpp
│       ├── ArborX_UnionFind.hpp
│       └── ArborX_WeightedEdge.hpp
├── distributed
│   ├── ArborX_DistributedTree.hpp
│   └── detail
│       ├── ArborX_DistributedTreeImpl.hpp
│       ├── ArborX_DistributedTreeNearest.hpp
│       ├── ArborX_DistributedTreeNearestHelpers.hpp
│       ├── ArborX_DistributedTreeSpatial.hpp
│       ├── ArborX_DistributedTreeUtils.hpp
│       └── ArborX_Distributor.hpp
├── geometry
│   ├── ArborX_Box.hpp
│   ├── ArborX_GeometryTraits.hpp
│   ├── ArborX_KDOP.hpp
│   ├── ArborX_Point.hpp
│   ├── ArborX_Ray.hpp
│   ├── ArborX_Segment.hpp
│   ├── ArborX_Sphere.hpp
│   ├── ArborX_Tetrahedron.hpp
│   ├── ArborX_Triangle.hpp
│   └── detail
│       └── ArborX_GeometryAlgorithms.hpp
├── interpolation
│   ├── ArborX_InterpMovingLeastSquares.hpp
│   └── detail
│       ├── ArborX_InterpDetailsCompactRadialBasisFunction.hpp
│       ├── ArborX_InterpDetailsMovingLeastSquaresCoefficients.hpp
│       ├── ArborX_InterpDetailsPolynomialBasis.hpp
│       └── ArborX_InterpDetailsSymmetricPseudoInverseSVD.hpp
├── kokkos_ext
│   ├── ArborX_KokkosExtAccessibilityTraits.hpp
│   ├── ArborX_KokkosExtArithmeticTraits.hpp
│   ├── ArborX_KokkosExtKernelStdAlgorithms.hpp
│   ├── ArborX_KokkosExtMinMaxReduce.hpp
│   ├── ArborX_KokkosExtSort.hpp
│   ├── ArborX_KokkosExtStdAlgorithms.hpp
│   ├── ArborX_KokkosExtUninitializedMemoryAlgorithms.hpp
│   ├── ArborX_KokkosExtVersion.hpp
│   └── ArborX_KokkosExtViewHelpers.hpp
├── misc
│   ├── ArborX_Containers.hpp
│   ├── ArborX_Exception.hpp
│   ├── ArborX_Heap.hpp
│   ├── ArborX_OperatorFunctionObjects.hpp
│   ├── ArborX_PriorityQueue.hpp
│   ├── ArborX_SortUtils.hpp
│   ├── ArborX_Stack.hpp
│   ├── ArborX_Utils.hpp
│   └── ArborX_Vector.hpp
└── spatial
    ├── ArborX_BruteForce.hpp
    ├── ArborX_CrsGraphWrapper.hpp
    ├── ArborX_LinearBVH.hpp
    └── detail
        ├── ArborX_AccessTraits.hpp
        ├── ArborX_AttachIndices.hpp
        ├── ArborX_BatchedQueries.hpp
        ├── ArborX_BruteForceImpl.hpp
        ├── ArborX_Callbacks.hpp
        ├── ArborX_CrsGraphWrapperImpl.hpp
        ├── ArborX_ExpandHalfToFull.hpp
        ├── ArborX_HalfTraversal.hpp
        ├── ArborX_HappyTreeFriends.hpp
        ├── ArborX_IndexableGetter.hpp
        ├── ArborX_MortonCode.hpp
        ├── ArborX_NearestBufferProvider.hpp
        ├── ArborX_NeighborList.hpp
        ├── ArborX_Node.hpp
        ├── ArborX_PairValueIndex.hpp
        ├── ArborX_PermutedData.hpp
        ├── ArborX_PredicateHelpers.hpp
        ├── ArborX_Predicates.hpp
        ├── ArborX_SpaceFillingCurves.hpp
        ├── ArborX_TraversalPolicy.hpp
        ├── ArborX_TreeConstruction.hpp
        ├── ArborX_TreeNodeLabeling.hpp
        ├── ArborX_TreeTraversal.hpp
        └── ArborX_TreeVisualization.hpp
```

**Comment by aprokop** [comment]
Couple builds failed for unrelated reasons.

**Comment by aprokop** [comment]
Had to rename two files, as we had both `ArborX_[Details]Dendrogram.hpp` and `ArborX_[Details]MinimumSpanningTree.hpp`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -73,18 +73,25 @@ cmake_dependent_option(ARBORX_ENABLE_GPU_AWARE_MPI
...
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/cluster>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/distributed>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/interpolation>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/interpolation/details>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/spatial>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
   $<INSTALL_INTERFACE:include/ArborX>
-  $<INSTALL_INTERFACE:include/ArborX/details>
   $<INSTALL_INTERFACE:include/ArborX/geometry>
+  $<INSTALL_INTERFACE:include/ArborX/geometry/detail>
```
Comment (dalg24) [inline CMakeLists.txt:84]:
```suggestion
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -73,18 +73,25 @@ cmake_dependent_option(ARBORX_ENABLE_GPU_AWARE_MPI
...
   $<INSTALL_INTERFACE:include/ArborX/kokkos_ext>
+  $<INSTALL_INTERFACE:include/ArborX/cluster>
+  $<INSTALL_INTERFACE:include/ArborX/cluster/detail>
```
Comment (dalg24) [inline CMakeLists.txt:87]:
```suggestion
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -73,18 +73,25 @@ cmake_dependent_option(ARBORX_ENABLE_GPU_AWARE_MPI
...
+  $<INSTALL_INTERFACE:include/ArborX/distributed>
+  $<INSTALL_INTERFACE:include/ArborX/distributed/detail>
```
Comment (dalg24) [inline CMakeLists.txt:89]:
```suggestion
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -73,18 +73,25 @@ cmake_dependent_option(ARBORX_ENABLE_GPU_AWARE_MPI
...
   $<INSTALL_INTERFACE:include/ArborX/interpolation>
-  $<INSTALL_INTERFACE:include/ArborX/interpolation/details>
+  $<INSTALL_INTERFACE:include/ArborX/interpolation/detail>
```
Comment (dalg24) [inline CMakeLists.txt:91]:
```suggestion
```

File: CMakeLists.txt [hunk 6]
```diff
@@ -73,18 +73,25 @@ cmake_dependent_option(ARBORX_ENABLE_GPU_AWARE_MPI
...
+  $<INSTALL_INTERFACE:include/ArborX/spatial>
+  $<INSTALL_INTERFACE:include/ArborX/spatial/detail>
```
Comment (dalg24) [inline CMakeLists.txt:93]:
```suggestion
```

**Review by dalg24 — APPROVED** [review]

**Pull request merged**